### ISSUE TITLE
 getDefiningElement NullPointerException

### ISSUE THREAD

**Issue by BanderMansourH** [wt-io-it/odoo-pycharm-plugin#43]
Plugin Version: 0.5.16
IDEA Version: PyCharm 2022.2

<details><summary>Unhandled exception in [StandaloneCoroutine{Cancelling}@69bebd6a, EDT]</summary>

```
java.lang.NullPointerException: Cannot invoke "java.util.function.Supplier.get()" because "exceptionSupplier" is null
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.getDefiningElement(OdooModelImpl.java:94)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.lambda$getDefiningElement$4(OdooModelImpl.java:65)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:928)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.getDefiningElement(OdooModelImpl.java:49)
	at at.wtioit.intellij.plugins.odoo.OdooGoToDeclarationHandler.getOdooModel(OdooGoToDeclarationHandler.java:63)
	at at.wtioit.intellij.plugins.odoo.OdooGoToDeclarationHandler.getGotoDeclarationTarget(OdooGoToDeclarationHandler.java:45)
	at com.intellij.codeInsight.navigation.actions.GotoDeclarationHandlerBase.getGotoDeclarationTargets(GotoDeclarationHandlerBase.java:14)
	...
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
	at com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:32)
	...
	at com.intellij.codeInsight.navigation.CtrlMouseHandler2$compute$2.invokeSuspend(CtrlMouseHandler.kt:221)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	...
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```
</details>

**Comment by ap-wtioit** [comment]
@BanderMansourH this seems to be the same issue as #30, do you have any additional information that would help me reproduce this?

**Comment by bigbear3001** [comment]
Reproduced when defining recursive module definitions with models in it

**Comment by bigbear3001** [comment]
New exception text, to be able to track down future errors faster:
Plugin Version: 0.6.9-SNAPSHOT
IntelliJ Version: PyCharm 2022.3
IntelliJ Build: PY-223.7571.203


<details><summary>at.wtioit.intellij.plugins.odoo.OdooPluginError: Cannot find defining elements for recursion.recursive with files [file:///home/bigbear3001/PycharmProjects/doodba-scaffolding/odoo/custom/src/private/recursion2/models/recursion.py, file:///home/bigbear3001/PycharmProjects/doodba-scaffolding/odoo/custom/src/private/recursion/models/recursion.py]</summary>

```
at.wtioit.intellij.plugins.odoo.OdooPluginError: Cannot find defining elements for recursion.recursive with files [file:///home/bigbear3001/PycharmProjects/doodba-scaffolding/odoo/custom/src/private/recursion2/models/recursion.py, file:///home/bigbear3001/PycharmProjects/doodba-scaffolding/odoo/custom/src/private/recursion/models/recursion.py]
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.lambda$getDefiningElement$6(OdooModelImpl.java:93)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.getDefiningElement(OdooModelImpl.java:93)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.lambda$getDefiningElement$4(OdooModelImpl.java:64)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:860)
	at at.wtioit.intellij.plugins.odoo.models.impl.OdooModelImpl.getDefiningElement(OdooModelImpl.java:48)
	at at.wtioit.intellij.plugins.odoo.search.AbstractOdooPsiElement.lambda$isValid$2(AbstractOdooPsiElement.java:217)
	at at.wtioit.intellij.plugins.odoo.WithinProject.call(WithinProject.java:15)
	at at.wtioit.intellij.plugins.odoo.search.AbstractOdooPsiElement.isValid(AbstractOdooPsiElement.java:216)
	at com.intellij.ide.actions.searcheverywhere.ml.features.statistician.SearchEverywhereSymbolStatistician$getContextName$$inlined$runReadAction$1.compute(actions.kt:62)
	...
	at com.intellij.ide.actions.searcheverywhere.MixedResultsSearcher$ContributorSearchTask.lambda$run$0(MixedResultsSearcher.java:183)
	at at.wtioit.intellij.plugins.odoo.search.OdooSEContributor.lambda$fetchWeightedElements$1(OdooSEContributor.java:85)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:847)
	at at.wtioit.intellij.plugins.odoo.search.OdooSEContributor.fetchWeightedElements(OdooSEContributor.java:78)
	at com.intellij.ide.actions.searcheverywhere.MixedResultsSearcher$ContributorSearchTask.run(MixedResultsSearcher.java:182)
	...
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:246)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	...
	at java.base/java.lang.Thread.run(Thread.java:833)
```
</details>

**Comment by bigbear3001** [comment]
this has been fixed in 0.6.9 (pending JetBrains approval). if you are eager to test the new version, you can use the jar file from https://github.com/wt-io-it/odoo-pycharm-plugin/tree/v0.6.9

**Comment by ap-wtioit** [comment]
0.6.9 has been approved and is available in the marketplace
